lib/projfs.c

```diff
@@ -428,13 +488,23 @@ static int projfs_op_symlink(char const *link, char const *path)
 static int projfs_op_create(char const *path, mode_t mode,
                             struct fuse_file_info *fi)
 {
+	/* FUSE sets fi.flags = O_CREAT | O_EXCL | O_WRONLY in fuse_lib_mknod,
+	 * untouched in fuse_lib_create where it comes straight from
+	 * FUSE_CREATE.  In Linux 4.9 at least, we never hit a codepath where
+	 * we send FUSE_CREATE without first checking that O_CREAT is set.
+	 * There's no guarantee O_EXCL (or O_TRUNC) are set, though, so we need
+	 * to hydrate it if it exists. */
+
```

Comment:
Awesome -- this kind of case-by-case flag testing was on my to-do list.  Thank you!